server/routerlicious/packages/lambdas/src/scribe/lambda.ts

```diff
@@ -589,8 +605,9 @@ export class ScribeLambda implements IPartitionLambda {
 		this.pendingP
 			.then(() => {
 				this.pendingP = undefined;
-				this.context.checkpoint(queuedMessage, this.restartOnCheckpointFailure);
-
+				if (!skipKafkaCheckpoint) {
+					this.context.checkpoint(queuedMessage, this.restartOnCheckpointFailure);
```

Comment:
Just to confirm, we skip kafka checkpointing here if the document is corrupted because we already checkpoint at document partition level?

Reply:
Yes, at the document partition level, we mark the document as corrupted and then checkpoint within the `markAsCorrupt()` function, which gets called once the error is thrown. If we try to checkpoint here, we get an error that does not let us continue to make the database checkpoint